src/ray/core_worker/task_manager.cc

```diff
@@ -968,7 +965,17 @@ bool TaskManager::RetryTaskIfPossible(const TaskID &task_id,
     spec = it->second.spec;
     num_retries_left = it->second.num_retries_left;
     num_oom_retries_left = it->second.num_oom_retries_left;
-    if (task_failed_due_to_oom) {
+    if (task_failed_due_to_stale_task) {
+      // Task failed due to stale task. This can only happen during actor unavailable
+      // and when you reconnect and retry, the actor already executed the last attempt
+      // which we consider as "ActorUnavailable" (attempt #1), and then the caller
+      // retries the task with the same seqno and got "StaleTask" (attempt #2). This
+      // attempt #2 does not consume a retry, though it does occupy an attempt number.
```

Comment:
it should not occupy an attempt number right? For end-user perspective, this retry is internal? what's the reason behind this?